Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ICriteriaExtensions.cs

```diff
@@ -31,21 +32,33 @@ public static class ICriteriaExtensions
             Junction whereJunction,
             IDictionary<string, object> parameters,
             string viewName,
-            string joinPropertyName,
+            string aggregateRootPropertyName,
+            string authViewPropertyName,
             string filterPropertyName,
             JoinType joinType,
             string authViewAlias = null)
         {
-            authViewAlias = string.IsNullOrWhiteSpace(authViewAlias)
-                ? $"authView{viewName}"
-                : $"authView{authViewAlias}";
+            string entityName = $"{viewName.GetAuthorizationViewClassName()}".GetFullNameForView();
+
+            if (viewName.ContainsIgnoreCase("EducationOrganizationIdToEducationOrganizationId"))
+            {
+                authViewAlias = string.IsNullOrWhiteSpace(authViewAlias) ? $"authTable{viewName}" : $"authTable{authViewAlias}";
```

Comment:
This refactored logic seems strange.  The only thing we seem to be changing is the passed in entityName, so why is the authViewAlias calculation also sprinkled around?  Can't we just default authViewAlias (based on existence of authViewAlias parameter), default entityName based on view, and just override the nHibernate name for EducationOrganizationToEducationOrganizationId to use this new GetFullNameForTable.

Reply:
I updated it now